### DISCUSSION TITLE
如何设置全局样式

### DISCUSSION THREAD

**Discussion by xmsz** [web-infra-dev/modern.js#3955] [Q&A] [answered]
背景
- 设置全局样式 比如reset
- 有多个页面

不知道该加在哪

**Comment by chenjiahan** [comment, marked as answer]
用 source.preEntry 配置：https://edenx.bytedance.net/configure/app/source/pre-entry.html#%E6%B7%BB%E5%8A%A0%E5%85%A8%E5%B1%80%E6%A0%B7%E5%BC%8F

**Reply by xmsz** [reply]
为什么我把
```
@tailwind base;
@tailwind components;
@tailwind utilities;
```
加到全局css里就没效果

**Reply by xmsz** [reply]
然后
```
import 'tailwindcss/base.css';
import 'tailwindcss/components.css';
import 'tailwindcss/utilities.css';
```
这个加在layout也没效果 只能放在页面文件或者页面css里

**Reply by targeral** [reply]
你说的效果是指你使用的 TailwindCss 的 classname 对应的样式没有生效嘛？

**Reply by xmsz** [reply]
> 你说的效果是指你使用的 TailwindCss 的 classname 对应的样式没有生效嘛？

我搞了一个DEMO 发现情况变复杂了 就是默认都很正常 但是加了一个插件(@douyinfe/semi-ui)以后
- 把tailwindCss的放在layout.tsx/global.css -> 组件的className失效
- 把tailwinCss放在page.tsx -> 都正常

demo地址：https://gitee.com/xmszer/demo-20230613 （github连不上太蛋疼）

**Reply by chenjiahan** [reply]
@xmsz 我看了下 .bg-red-500 这些类名都生效的，你指的是 ml-3 没生效吗？看起来 ml-3 是有的，只是被组件的样式覆盖了：

![image](https://github.com/web-infra-dev/modern.js/assets/7237365/9f097a9d-0073-41d7-a87a-bd00618724db)

所以这是一个 CSS 样式引入顺序的问题，而不是 tailwindcss 没生效，你调整下 CSS 引入顺序就可以了。

**Reply by xmsz** [reply]
我居然想复杂了 确实样式覆盖了
